server/model/enum.go

```diff
@@ -28,6 +28,14 @@ const (
 	Retries                DeploymentMetric = "Retries"
 )
 
+type TelemetryStatus string
+
+const (
+	TelemetryFailed   TelemetryStatus = "failed"
```

Comment:
These names are misleading, they seem to indicate status of telemetry as if the telemetry was failing, succeeding or being cancelled. 
Better names would include the name of what has failed, succeeded or was cancelled, which in this case is the "deployment". With that in mind, this would look better and be more obvious:
```go
DeploymentFailed TelemetryStatus = "failed"
DeploymentSucceeded TelemetryStatus = "succeeded"
DeploymentCanceled TelemetryStatus = "canceled"
```
Also notice that in the code snippet here, the successful deployment the name ends with "...Succeeded" to be consistent with the other ones.

Reply:
Yeah, that make sense.. will change that after we agree on the function name above.